pkg/proxy/winkernel/proxier_test.go

```diff
@@ -205,22 +222,30 @@ func TestCreateRemoteEndpointOverlay(t *testing.T) {
 					IP: epIpAddressRemote,
 				}},
 				Ports: []v1.EndpointPort{{
-					Name: svcPortName.Port,
-					Port: int32(svcPort),
+					Name:     svcPortName.Port,
+					Port:     int32(svcPort),
+					Protocol: v1.ProtocolTCP,
 				}},
 			}}
 		}),
 	)
-
+	proxier.setInitialized(true)
 	proxier.syncProxyRules()
 
-	if proxier.endpointsMap[svcPortName][0].hnsID != guid {
-		t.Errorf("%v does not match %v", proxier.endpointsMap[svcPortName][0].hnsID, guid)
+	ep := proxier.endpointsMap[svcPortName][0]
+	epInfo, ok := ep.(*endpointsInfo)
+	if !ok {
+		t.Errorf("Failed to cast endpointsInfo %q", svcPortName.String())
+
+	} else {
+		if epInfo.hnsID != guid {
+			t.Errorf("%v does not match %v", epInfo.hnsID, guid)
```

Comment:
Maybe add something to make it clearer which field is not matching:

```suggestion
			t.Errorf("Unexpected value for hnsID, got: %v, expected %v", epInfo.hnsID, guid)
```